File: app/src/main/java/com/example/testapplication/orientationapp/pin/Pin.java
package com.example.testapplication.orientationapp.pin;

import android.content.Context;
import android.content.SharedPreferences;


public class Pin {
    private static Pin instance;
    private String pin = "";
    private boolean pinEnabledForApp;
    private boolean skipCheck;
    private SharedPreferences prefs;

    public static void destroy() {
        instance = null;
    }

    public static synchronized Pin getInstance(Context ctx) {
        if (instance == null) {
            instance = new Pin(ctx);
        }
        return instance;
    }

    private Pin(Context ctx) {
        prefs = ctx.getSharedPreferences("prefs", Context.MODE_PRIVATE);
        pin = prefs.getString("pin", "");
        pinEnabledForApp = prefs.getBoolean("pinEnabledForApp", false);
        skipCheck = prefs.getBoolean("skipCheck", false);
    }

    public boolean isEnabledForApp() {
        return pinEnabledForApp;
    }

    public String getPin() {
        return pin;
    }

    public void setPin(String pin) {
        prefs.edit().putString("pin", pin).apply();
        this.pin = pin;
    }

    public boolean comparePin(String pin) {
        return this.pin.equals(pin);
    }

    public void setPinEnabledForApp(boolean enabled) {
        prefs.edit().putBoolean("pinEnabledForApp", enabled).apply();
        pinEnabledForApp = enabled;
    }

    public void setSkipCheck(boolean enabled) {
        if (pin.isEmpty()) {
            return;
        }
        skipCheck = enabled;
        prefs.edit().putBoolean("skipCheck", enabled).apply();

    }

    public boolean isSkipCheck() {
        return skipCheck;
    }

    public boolean isPinSet() {
        return !prefs.getString("pin", "").isEmpty();
    }

    public void reset() {
        SharedPreferences.Editor editor = prefs.edit();
        editor.remove("pin");
        editor.remove("pinEnabledForApp");
        editor.remove("skipCheck");
        editor.apply();
    }
}

File: app/src/main/java/com/example/testapplication/orientationapp/util/Pin.kt
package com.example.orientation.orientationapp.util

import android.content.Context
import android.content.SharedPreferences

class Pin private constructor(ctx: Context) {
    private var pin: String? = ""
    var isEnabledForApp: Boolean
        private set
    private var skipCheck: Boolean
    private val prefs: SharedPreferences
    fun getPin(): String? {
        return pin
    }

    fun setPin(pin: String?) {
        prefs.edit().putString("pin", pin).apply()
        this.pin = pin
    }

    fun comparePin(pin: String): Boolean {
        return this.pin == pin
    }

    fun setPinEnabledForApp(enabled: Boolean) {
        prefs.edit().putBoolean("pinEnabledForApp", enabled).apply()
        isEnabledForApp = enabled
    }

    fun setSkipCheck(enabled: Boolean) {
        if (pin!!.isEmpty()) {
            return
        }
        skipCheck = enabled
        prefs.edit().putBoolean("skipCheck", enabled).apply()
    }

    fun isSkipCheck(): Boolean {
        return skipCheck
    }

    val isPinSet: Boolean
        get() = !prefs.getString("pin", "")!!.isEmpty()

    fun reset() {
        val editor = prefs.edit()
        editor.remove("pin")
        editor.remove("pinEnabledForApp")
        editor.remove("skipCheck")
        editor.apply()
    }

    companion object {
        private var instance: Pin? = null
        fun destroy() {
            instance = null
        }

        @Synchronized
        fun getInstance(ctx: Context): Pin? {
            if (instance == null) {
                instance = Pin(ctx)
            }
            return instance
        }
    }

    init {
        prefs = ctx.getSharedPreferences("prefs", Context.MODE_PRIVATE)
        pin = prefs.getString("pin", "")
        isEnabledForApp = prefs.getBoolean("pinEnabledForApp", false)
        skipCheck = prefs.getBoolean("skipCheck", false)
    }
}